engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/EqualsMultiValueTest.java

```diff
@@ -0,0 +1,268 @@
+    executeInContext(
+        context,
+        () -> {
+          var builtins = ContextUtils.leakContext(context).getBuiltins();
+          var intType = builtins.number().getInteger();
+          var textType = builtins.text();
+          var bothTypes = new Type[] {textType, intType};
+
+          var text = Text.create("Hi");
+          var ahoj = Text.create("Ahoj");
+          var integer = 4L;
+          //
+          // following variable represents result of
+          //     x = _ : (Text & Integer) : Text
+          // e.g. multi value with Text and Integer, casted to Text only
+          //
+          var multiV = EnsoMultiValue.create(bothTypes, 1, text, integer);
+
+          assertTrue("'Hi' == multiV", equalityCheck(text, multiV));
+          assertFalse("'Ahoj' != multiV", equalityCheck(ahoj, multiV));
+          assertFalse(
+              "Don't consider extra Integer type in equals", equalityCheck(integer, multiV));
+          assertFalse("5 != t4", equalityCheck(5L, multiV));
+          assertFalse(
+              "Don't consider extra Integer type in equals", equalityCheck(multiV, integer));
+          assertFalse("4 != t5", equalityCheck(multiV, 5L));
+          assertTrue("multiV == 'Hi'", equalityCheck(multiV, text));
+          assertFalse("multiV != 'Ahoj'", equalityCheck(multiV, ahoj));
+
+          return null;
+        });
+  }
+
+  @Test
+  public void testEqualityIntegerAndMultiValueWithBoth() {
+    executeInContext(
+        context,
+        () -> {
+          var builtins = ContextUtils.leakContext(context).getBuiltins();
+          var intType = builtins.number().getInteger();
+          var textText = builtins.text();
+          var fourExtraText =
+              EnsoMultiValue.create(
+                  new Type[] {textText, intType}, 2, new Object[] {Text.create("Hi"), 4L});
+
+          assertTrue("4 == 4t", equalityCheck(4L, fourExtraText));
+          assertFalse("5 != 4t", equalityCheck(5L, fourExtraText));
+          assertTrue("4t == 4", equalityCheck(fourExtraText, 4L));
+          assertFalse("4t != 5", equalityCheck(fourExtraText, 5L));
+
+          return null;
+        });
+  }
+
+  @Test
+  public void testEqualityIntegerAndMultiValueWithIntText() {
+    executeInContext(
+        context,
+        () -> {
+          var builtins = ContextUtils.leakContext(context).getBuiltins();
+          var intType = builtins.number().getInteger();
+          var textText = builtins.text();
+          var fourExtraText =
+              EnsoMultiValue.create(
+                  new Type[] {intType, textText}, 2, new Object[] {4L, Text.create("Hi")});
+
+          assertTrue("4 == 4t", equalityCheck(4L, fourExtraText));
+          assertFalse("5 != 4t", equalityCheck(5L, fourExtraText));
+          assertTrue("4t == 4", equalityCheck(fourExtraText, 4L));
+          assertFalse("4t != 5", equalityCheck(fourExtraText, 5L));
+
+          return null;
+        });
+  }
```

Comment:
This test seems almost identical to the one above, maybe we can extract a common part?

Reply:
Right, finding difference isn't simple, but reading, debugging, changing a single test is simpler when the code is duplicated.